sdcm/provision/aws/provisioner.py

```diff
@@ -0,0 +1,290 @@
+                instance_parameters=instance_parameters,
+                count=count,
+                tags=tags,
+            )
+        raise RuntimeError("Unexpected provision_parameters type")
+
+    @staticmethod
+    def _is_provision_type_fleet(count: int) -> bool:
+        return count > SPOT_CNT_LIMIT
+
+    def _provision_instance_limit(self, count: int) -> int:
+        return SPOT_FLEET_LIMIT if self._is_provision_type_fleet(count) else SPOT_CNT_LIMIT
+
+    @property
+    def _spot_valid_until(self) -> datetime.datetime:
+        return datetime.datetime.now() + datetime.timedelta(minutes=SPOT_REQUEST_TIMEOUT / 60 + 5)
+
+    @staticmethod
+    def _ec2_client(provision_parameters: ProvisionParamsBase) -> EC2Client:
+        return ec2_clients[provision_parameters.region_name]
+
+    @staticmethod
+    def _full_availability_zone_name(provision_parameters: ProvisionParamsBase) -> str:
+        return provision_parameters.region_name + provision_parameters.availability_zone
+
+    @staticmethod
+    def _spot_block_duration(provision_parameters: ProvisionParamsBase) -> Optional[int]:
+        """
+        A period in minutes AWS will hold spot instance for us.
+        AWS have limit on it, if test is going to be run
+        """
+        if provision_parameters.duration is None:
+            return None
+        if provision_parameters.duration >= MAX_SPOT_DURATION_TIME:
+            LOGGER.debug("Requested stop time more than AWS limit (360 minutes). Spot duration is going to be ignored")
+            return None
+        return int(provision_parameters.duration)
+
+    @staticmethod
+    def _provision_guaranteed(
+            provision_parameters: ProvisionParamsBase,
+            instance_parameters: AWSInstanceParams,
+            count: int,
+            tags: List[TagsType]) -> List[Instance]:
+        instance_parameters_dict = instance_parameters.dict(
+            exclude_none=True, exclude_defaults=True, exclude_unset=True)
+        LOGGER.debug("[%s] Creating {count} on-demand instances using AMI id '%s' with following parameters:\n%s",
+                     provision_parameters.region_name,
+                     instance_parameters.ImageId,
+                     instance_parameters_dict
+                     )
+        instances = ec2_services[provision_parameters.region_name].create_instances(
+            **instance_parameters_dict, MinCount=count, MaxCount=count)
+        LOGGER.debug("Created instances: %s.", instances)
+        if instances:
+            for ind, instance in enumerate(instances):
+                instance_tags = tags.pop()
+                set_tags_on_instances(
+                    region_name=provision_parameters.region_name,
+                    instance_ids=[instance.instance_id for instance in instances],
+                    tags={'Name': 'spot_fleet_{}_{}'.format(instance.instance_id, ind)} | instance_tags,
+                )
+        return instances
+
+    def _provision_not_guaranteed(
```

Comment:
Why "not_gauranteed" instead of spot? This is the term that AWS using

Reply:
The idea here is develop language that is apart of the api.
This particular method is supposed to process guaranteed requests